packages/flutter_markdown/lib/src/widget.dart

```diff
@@ -36,7 +36,7 @@ typedef MarkdownTapLinkCallback = void Function(
 ///
 /// Used by [MarkdownWidget.imageBuilder]
 typedef MarkdownImageBuilder = Widget Function(
-    Uri uri, String? title, String? alt);
+    Uri uri, String? title, String? alt, double? width, double? height);
```

Comment:
Is there a way of adding these optional arguments in a way that doesn't induce an API breaking change?

Reply:
I don't think, the only solution is mark this builder as deprecated and add new builder with image size

```dart
@Deprecated('Use sizedImageBuilder instead.')
final MarkdownImageBuilder? imageBuilder;
```

the new builder :

```dart
typedef MarkdownSizedImageBuilder = Widget Function(
    Uri uri, String? title, String? alt, MarkdownImageSize? size);
```

the MarkdownImageSize

```dart
class MarkdownImageSize {
  final double? width;
  final double? height;

  const MarkdownImageSize({this.width, this.height});
}
```

Reply:
@stuartmorgan are you means use all fields (uri, alt, title and size) as class parameter on new builder or only size as MarkdownImageSize class...

Reply:
It doesn't necessarily need to have all fields, but it needs to be extensible. It should be able to answer the question "how would we add a new parameter in the future without a breaking change?"